doc/educational_decoder/harness.c

```diff
@@ -50,6 +50,7 @@ static buffer_s read_file(const char *path)
 
     fclose(f);
     buffer_s const b = { ptr, size };
+    free(ptr);
```

Comment:
This change is incorrect :
the whole point of this function is to return a populated buffer, passed as a member of the `buffer_s` structure, effectively transferring ownership to the caller of the function (which will have to free it later, using the provided `freeBuffer()` function).

Maybe this could be documented if it's not clear enough.